### ISSUE TITLE
``no_std`` environment is unusable

### ISSUE THREAD

**Issue by Vickerinox** [Frostie314159/ieee80211-rs#1]
When using the crate with a ``no_std`` environment ``either`` from the dependency ``itertools`` still asks for std, contradicting the readme.

compiled with rust toolchain ``nightly-2024-02-20`` on windows

error
```
error[E0463]: can't find crate for `std`                                                   
  --> C:\Users\----\.cargo\registry\src\index.crates.io-6f17d22bba15001f\either-1.10.0\src\lib.rs:19:1
   |
19 | extern crate std;
   | ^^^^^^^^^^^^^^^^^ can't find crate
   |
```

cargo.toml dependencies
```toml
[dependencies]
ieee80211 = {version = "0.1.3", default-features = false }
scroll = { version = "0.12.0", default-features = false }
```

**Comment by Frostie314159** [comment]
Oh, thanks for pointing this out, I'm going to look into it right away.

**Comment by Frostie314159** [comment]
Fixed with `c3500af`. Thanks a lot! `itertools` wasn't actually necessary anymore and removing it resolved the issue. I also introduced a test, which only compiles if `no_std` is operational.
